src/metadata/resolution.rs

```diff
@@ -57,37 +26,27 @@ impl<T> TryFrom<&mut Decoder<T>> for Resolution
 where
     T: Read + Seek,
 {
-    type Error = ResolutionError;
+    type Error = TiffError;
 
     // Extract resolution metadata from a TIFF decoder
-    fn try_from(decoder: &mut Decoder<T>) -> Result<Self, ResolutionError> {
+    fn try_from(decoder: &mut Decoder<T>) -> Result<Self, TiffError> {
         // Parse resolution unit
         let unit = decoder
             .get_tag(tiff::tags::Tag::ResolutionUnit)
-            .map_err(|_| ResolutionError::MissingProperty {
+            .map_err(|_| TiffError::MissingPropertyError {
                 name: "Resolution Unit",
             })?
-            .into_u16()
-            .map_err(|s| ResolutionError::InvalidResolutionUnit {
-                source: Box::new(s),
-            })?;
-        let unit = ResolutionUnit::from_u16(unit)
-            .ok_or(ResolutionError::CastResolutionUnit { val: unit })?;
+            .into_u16()?;
+        let unit = ResolutionUnit::from_u16(unit).ok_or(TiffError::Other {
+            message: "invalid resolution unit".to_string(),
+        })?;
 
         // Parse x resolution
-        let x_resolution = decoder
-            .get_tag_u32_vec(tiff::tags::Tag::XResolution)
-            .map_err(|s| ResolutionError::InvalidResolution {
-                source: Box::new(s),
-            })?;
+        let x_resolution = decoder.get_tag_u32_vec(tiff::tags::Tag::XResolution)?;
```

Comment:
Maybe we could improve the variable names here to make the code easier to understand? Something like this:
```rust
let [pixels, inches] = ...
let x_ppi = pixels / inches  // This is just an example... I don't actually know what values are being used in this code.
```

Reply:
I cleaned this up a bit. The TIFF encoder stores this field as a rational number, so the parsed values are numerator and denominator that are combined to produce a final float